### PR TITLE
v1.4.0rc2

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #<issue_number>

## Before submitting
- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)
- [ ] Did you list all the **breaking changes** introduced by this pull request?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGMT !

File: CHANGELOG.md [hunk 1]
```diff
@@ -249,317 +122,119 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
 - Fixed a bug where `precision=64` with `accelerator='ddp_spawn'` would throw a pickle error ([#6924](https://github.com/PyTorchLightning/pytorch-lightning/pull/6924))
-
-
-- Do not override the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))
-
-
-- Support manual optimization with DeepSpeed ([#7970](https://github.com/PyTorchLightning/pytorch-lightning/pull/7970))
-
-
+- Fixed not override the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))
```
Comment (awaelchli) [inline CHANGELOG.md:203]:
```suggestion
- Fixed overriding the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -249,317 +122,119 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+- Fixed support manual optimization with DeepSpeed ([#7970](https://github.com/PyTorchLightning/pytorch-lightning/pull/7970))
```
Comment (awaelchli) [inline CHANGELOG.md:204]:
```suggestion
- Fixed support for manual optimization with DeepSpeed ([#7970](https://github.com/PyTorchLightning/pytorch-lightning/pull/7970))
```

**Review by awaelchli — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (carmocca) [reply CHANGELOG.md:203]: Not a fan of renaming the CHANGELOG entries. Starting every phrase with "Fixed ..." is limiting.

The new sentence is confusing as now it sounds like overriding the epoch value wasn't working and now we DO it properly

When it's the opposite.

Comment (awaelchli) [reply CHANGELOG.md:203]: I'm just correcting the grammar here on top of the existing diff :)

Comment (carmocca) [reply CHANGELOG.md]: I'm referring to the change that @Borda made:

from:

`- Do not override the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))`

to

`- Fixed overriding the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))`

In the changed version, it's not clear what the fix is.

Comment (Borda) [reply CHANGELOG.md]: Sorry for that =/

File: CHANGELOG.md [hunk 3]
```diff
@@ -249,317 +122,119 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
 - Fixed DeepSpeed not properly setting the trainer `lr_schedulers` attribute ([#8527](https://github.com/PyTorchLightning/pytorch-lightning/pull/8527))
-
-
 - Fixed experiment version and log-dir divergence in DDP when using multiple `Trainer` instances in sequence ([7403](https://github.com/PyTorchLightning/pytorch-lightning/pull/7403))
-
-
-- Enabled manual optimization for TPUs ([#8458](https://github.com/PyTorchLightning/pytorch-lightning/pull/8458))
-
-
+- Fixed Enabling manual optimization for TPUs ([#8458](https://github.com/PyTorchLightning/pytorch-lightning/pull/8458))
```
Comment (Borda) [inline CHANGELOG.md:232]:
```suggestion
- Enabled manual optimization for TPUs ([#8458](https://github.com/PyTorchLightning/pytorch-lightning/pull/8458))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -249,317 +122,119 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
-
-
-- Do not override the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))
-
-
-- Support manual optimization with DeepSpeed ([#7970](https://github.com/PyTorchLightning/pytorch-lightning/pull/7970))
-
-
+- Fixed overriding the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))
+- Fixed support for manual optimization with DeepSpeed ([#7970](https://github.com/PyTorchLightning/pytorch-lightning/pull/7970))
```
Comment (Borda) [inline CHANGELOG.md:204]:
```suggestion
- Support for manual optimization with DeepSpeed ([#7970](https://github.com/PyTorchLightning/pytorch-lightning/pull/7970))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -249,317 +122,119 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
 - Fixed a bug where `precision=64` with `accelerator='ddp_spawn'` would throw a pickle error ([#6924](https://github.com/PyTorchLightning/pytorch-lightning/pull/6924))
-
-
-- Do not override the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))
-
-
-- Support manual optimization with DeepSpeed ([#7970](https://github.com/PyTorchLightning/pytorch-lightning/pull/7970))
-
-
+- Fixed overriding the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))
```
Comment (Borda) [inline CHANGELOG.md:203]:
```suggestion
- Do not override the existing `epoch` value in `logged_metrics` when already logged by the user ([#7982](https://github.com/PyTorchLightning/pytorch-lightning/issues/7982))
```

**Review by carmocca — APPROVED** [review]
Thanks!
